tests/pivots/spreadsheet_pivot/spreadsheet_pivot.test.ts

```diff
@@ -446,6 +446,23 @@ describe("Spreadsheet Pivot", () => {
     expect(getEvaluatedGrid(model, "B26:F26")).toEqual([["5", "9", "14", "Total", ""]]);
   });
 
+  test("Date dimensions should support empty cells", () => {
+    const model = createModelWithPivot("A1:I5");
+    updatePivot(model, "1", {
+      columns: [{ name: "Created on", granularity: "month_number", order: "asc" }],
+      rows: [],
+      measures: [{ name: "Expected Revenue", aggregator: "sum" }],
+    });
+    setCellContent(model, "A26", `=pivot(1)`);
+    expect(getEvaluatedGrid(model, "B26:F26")).toEqual([
+      ["February", "March", "April", "Total", ""],
+    ]);
+    setCellContent(model, "A4", ""); //The only cell in the range with month = February
```

Comment:
typically what we discussed: create a pivot with lots of things, then delete stuff that we don't want :smile: 
this makes the test convoluted and hard to read.

not longer:
```ts
  test("Date dimensions should support empty cells", () => {
    const grid = {
      A1: "Date",
      A2: "",
      A3: "2024-03-01",
      A4: "=pivot(1)",
    }
    const model = createModelFromGrid(grid);
    addPivot(model, "A1:A3", {
      columns: [{ name: "Date", granularity: "month_number" }],
      measures: [{ name: "__count" }],
    });
    expect(getEvaluatedGrid(model, "B4:E4")).toEqual([
      ["March", "(Undefined)", "Total", ""],
    ]);
  });
```

Reply:
I fully agree.